lib/magma/server/query.rb

```diff
@@ -6,10 +6,10 @@ def action
     return failure(401, errors: [ 'You are unauthorized' ]) unless @user && @user.can_view?(@project_name)
     begin
       return success(Magma::Predicate.to_json, 'application/json') if @params[:query] == '::predicates'
-      question = Magma::Question.new(@project_name, @params[:query])
+      question = Magma::Question.new(@project_name, @params[:query], :timeout => 300000)
       return_data = {answer: question.answer, type: question.type}
       return success(return_data.to_json, 'application/json')
-    rescue ArgumentError => e
+    rescue StandardError => e
```

Comment:
Does the timeout generate a more specific error? I would probably catch that rather than catching StandardError, since not every StandardError implies a client error. In this case I'm not sure a timeout is a 422, it should probably be something like a 500...

Reply:
How do you feel about catching the DatabaseError and sending a 500? I don't think a 504 is appropriate  because other things can raise a DatabaseError.

Reply:
Yeah in general I feel like 500 is all they need to know. Definitely not a client error, they just submitted a well-formulated query. It's our fault we can't fulfill it, so probably a 501 'Not Implemented'